### PR TITLE
feat: storing index for code first migration

### PR DESCRIPTION
**Pull request by laljikanjareeya**

### CODE REVIEW

**Review by olavloite — CHANGES_REQUESTED** [review]
Good spot that we missed support for this feature!

I however don't think we should use a custom annotation for this. `STORING` is not a Spanner specific feature, but something that is supported by many relational databases. It also seems to be supported by the standard EF Core implementation: https://docs.microsoft.com/en-us/ef/core/modeling/indexes?tabs=data-annotations#included-columns

**Comment by laljikanjareeya** [comment]
It seems like `IncludeProperties` are extension method of `IndexBuilder` in `Microsoft.EntityFrameworkCore.SqlServer` library.
Not available in EF Core.

**Comment by olavloite** [comment]
> It seems like `IncludeProperties` are extension method of `IndexBuilder` in `Microsoft.EntityFrameworkCore.SqlServer` library.
> Not available in EF Core.

Hmmm... That's true, although it's not really clear from the documentation.

Could we implement this in the same way for Spanner as for SQL Server? That is:
1. The keyword should be `Storing` and not `IsStoring`.
2. The property should accept a collection of columns and not only one.

**Comment by laljikanjareeya** [comment]
> Could we implement this in the same way for Spanner as for SQL Server? That is:
> 
> 1. The keyword should be `Storing` and not `IsStoring`.
> 2. The property should accept a collection of columns and not only one.

Done. PTAL.

File: Google.Cloud.EntityFrameworkCore.Spanner.Tests/Migrations/MigrationSqlGeneratorTestBase.cs [hunk 1]
```diff
@@ -387,6 +387,28 @@ public virtual void CreateIndexOperation_is_unique()
                 });
         }
 
+        public virtual void CreateIndexOperation_is_storing()
```
Comment (olavloite) [inline Google.Cloud.EntityFrameworkCore.Spanner.Tests/Migrations/MigrationSqlGeneratorTestBase.cs:390]: nit:
```suggestion
        public virtual void CreateIndexOperation_storing()
```

File: Google.Cloud.EntityFrameworkCore.Spanner.Tests/Migrations/MigrationSqlGeneratorTestBase.cs [hunk 2]
```diff
@@ -387,6 +387,28 @@ public virtual void CreateIndexOperation_is_unique()
...
+                modelBuilder => modelBuilder.Entity("Albums").Property<string>("AlbumTitle"),
+                new CreateIndexOperation
+                {
+                    Name = "AlbumsByAlbumTitle2",
+                    Table = "Albums",
+                    Columns = new[] { "AlbumTitle" },
+                    [SpannerAnnotationNames.Storing] = new string[] { "MarketingBudget" }
+                });
+
+        public virtual void CreateIndexOperation_is_storing_with_multiple_column()
```
Comment (olavloite) [inline Google.Cloud.EntityFrameworkCore.Spanner.Tests/Migrations/MigrationSqlGeneratorTestBase.cs:401]: nit:
```suggestion
        public virtual void CreateIndexOperation_storing_with_multiple_columns()
```

File: Google.Cloud.EntityFrameworkCore.Spanner/Migrations/SpannerMigrationsSqlGenerator.cs [hunk 3]
```diff
@@ -72,6 +72,23 @@ public SpannerMigrationsSqlGenerator(MigrationsSqlGeneratorDependencies dependen
...
+                for (var i = 0; i < storingColumns.Count; i++)
+                {
+                    builder.Append(Dependencies.SqlGenerationHelper.DelimitIdentifier(storingColumns[i]));
+
+                    if (i != storingColumns.Count - 1)
+                    {
+                        builder.Append(", ");
+                    }
+                }
+
```
Comment (olavloite) [inline Google.Cloud.EntityFrameworkCore.Spanner/Migrations/SpannerMigrationsSqlGenerator.cs:88]:
```suggestion
                .Append(ColumnList(storingColumns))
```

**Review by olavloite — APPROVED** [review]
A couple of nits, but otherwise LGTM.

**Pull request merged**